### ISSUE TITLE
Correct velocity dispersion for instrumental resolution effects as option in plotting

### ISSUE THREAD

**Issue by KateRowlands** [sdss/marvin#175]
Correct velocity dispersion for instrumental resolution effects as an option when plotting maps using marvin.tools.map.

**Comment by havok2063** [comment]
We may want to consider having Marvin apply these corrections, so people don't have to.  STELLAR_SIGMA, and EMLINE_GSIGMA. 

From the DAP TRM. 
```

Corrections to apply:

Note that the stellar and gas velocity dispersions must be corrected for instrumental resolution effects.

The corrected gas velocity dispersion is:

sigma_gas_corr = sqrt( EMLINE_GSIGMA2 - EMLINE_INSTSIGMA2 )

The corrected stellar velocity dispersion is:

sigma_star_corr = sqrt( STELLAR_SIGMA2 - STELLAR_SIGMACORR2 )

In both cases, beware of imaginary numbers. That is, when the correction is larger than the provided value, the above equations result in taking the sqrt of a negative number.
```

**Comment by kbwestfall** [comment]
I think Kate means just do the quadrature difference sqrt(sigma_obs^2-sigma_corr^2) when plotting the dispersion map.  I’m not sure I would do this.  I think we want to allow people to create new maps based on some algebraic combination of other maps, but I think adding this feature specifically to Maps will break the generality of the Maps class.  In any case, this is a feature request, not a bug.

**Comment by havok2063** [comment]
So maybe an option to do so if people desire?

**Comment by kbwestfall** [comment]
To me this is like adding the `getMapRatio()` function when you're creating the BPT maps.  You could have a functions like `getMapDifference(quadrature=True)`, `getMapSum()`, `getSqrtMap()`, etc.  Or better yet, define the '+', etc symbols in the Map class...

**Comment by havok2063** [comment]
Ah yeah, generic map algebra has been something on our minds to do.

**Comment by bretthandrews** [comment]
I like it! Let's override `+`, `-`, `*`, `/`, and `**` so that we can do general map arithmetic!

How about we add a warning if people try to get a `stellar_sigma` or `emline_gsigma` Map that points out the needed corrections and how to do them?

**Comment by kbwestfall** [comment]
Yeah, I think the warning is a good short-term solution.  In fact, because it's easy to do and people love warnings, we should brainstorm a set of warnings based on the known issues in the DRP/DAP data that we should put into Marvin so that people see them often.  Well, I'm joking about the "love warnings" part, but I'm serious about the brainstorm part.  Maybe not for this release though...

**Comment by havok2063** [comment]
Yeah, not essential for 2.1.  We're already 2 days behind schedule according to our personal Milestone, as Github keeps reminding me. :(

**Comment by bretthandrews** [comment]
I agree with that decision if Kyle doesn't think we should change it. Though I imagine that people will misuse it, and we'll still get blamed.

**Comment by kbwestfall** [comment]
Yeah, I know, but hopefully they'll appreciate that we all work at pace and that we can't satisfy all their requests.  It was actually a conscious decision on our part to keep the measurements and the corrections separate so that users would have to do the corrections, and understand the subtleties of those corrections, themselves.  Maybe this approach is a bit heavy handed, but I think it does go some way toward making sure users understand how they should handle/use the data.

**Comment by bretthandrews** [comment]
Fixed in #430 

```python
maps = Maps('8485-1901')
stsig = maps.stellar_sigma
stsig_corrected = stsig.inst_sigma_correction()
stsig_corrected.plot()
```
